std/range/package.d

```diff
@@ -2012,6 +2012,13 @@ if (isInputRange!(Unqual!Range) &&
         return _maxAvailable == 0 || source.empty;
     }
 
+    static if (is(typeof((cast(const R) source).empty)))
+        /// ditto
+        @property bool empty() const
+        {
+            return _maxAvailable == 0 || source.empty;
+        }
+
```

Comment:
what's with this code?

Reply:
This allow making `empty` of `substitute` `const`, but imho inference of `const` is something that we should fix on a global level.